### ISSUE TITLE
Unique widgets, part of an interface

### ISSUE THREAD

**Issue by zerline** [jupyter-widgets/ipywidgets#2575]
Hi
I've been working on additional HTML attributes and methods for ipywidgets that are useful when you want to build interfaces with them: `focus` and `tabindex`.
In an ipywidgets based interface, we may want to `focus` or `blur` programmatically  a widget, for example a `run` button.
And we may want to control whether a specific interface component can receive focus, and sometimes also in which order that focus occurs.
You can see my attempts on Sage Combinat Widgets, branch `develop`:
https://github.com/sagemath/sage-combinat-widgets/blob/develop/unit_widgets/unit_widgets.py
https://github.com/sagemath/sage-combinat-widgets/blob/develop/js/src/unit_widgets.ts

**Comment by zerline** [comment]
NB : This subpackage has been renamed to "singleton widgets" and it is still part (for now) of Sage Combinat Widgets:

 https://github.com/sagemath/sage-combinat-widgets/blob/develop/singleton_widgets/singleton_widgets.py

**Comment by zerline** [comment]
Two questions here:

1) would anyone be interested in a PR for all, or part of, these features?
* tooltips on more widgets (Dropdown, ToggleButton ..)
* setting focus on a given widget
* setting tabindex, in particular include, or not, a given widget in the list of focusable elements

2) If you have a look at my code, there is a lot of duplicate lines. Is there a solution to this problem (apart from integrating those lines into ipywidgets at the right hierarchical level!)

**Comment by vidartf** [comment]
All of these points sound interesting, maybe with some reservations about the focus point. I haven't yet looked at the code, so seeing one or more PRs could be very helpful. 👍

**Comment by zerline** [comment]
Hi @vidartf 
I will find more time for PRs in tow weeks time.
About focus, I guess your reservation comes from the fact it will only work for actual widget singletons?

**Comment by vidartf** [comment]
If by singletons you mean "widgets with only one view", then yes 😃 Some other concerns as well.

**Comment by zerline** [comment]
Yes @vidartf absolutely. These need to be that kind of "frontend singletons" (*) if you want to programmatically set focus. Normally, widgets that are part of a larger interface are not duplicated (at least not under the same name). 
(*) do you have a better naming suggestion?

**Comment by zerline** [comment]
Cf PRs
* https://github.com/jupyter-widgets/ipywidgets/pull/2626 (tabindex)
* https://github.com/jupyter-widgets/ipywidgets/pull/2628 (tooltip)
* https://github.com/jupyter-widgets/ipywidgets/pull/2630 (focus)
